### PR TITLE
Report table to query solution

### PR DESCRIPTION
**Pull request by yashcodes**
Solves [#2191](https://github.com/JuliaOpt/JuMP.jl/issues/2191)
This PR adds a function `result!(model)` whose purpose is to structure the result obtained from the solver in a proper Tabular form, this function still have lots of possibilities to extend that's why we can call it as a basic version of `result!` like the accounting of multiple numbers of results is still pending, accessing MOI attribute likes RelativeGap(), SimplexIterations() etc.
From the current version of `result!` we can access dual solutions by setting keyword argument `dual_query` equals to `true` (`result!(model, dual_query = true)` )

### CODE REVIEW

**Comment by yashcodes** [comment]
@odow @mtanneau @blegat 
Can anyone review this PR?

**Comment by mtanneau** [comment]
I'm not sure I understand the motivation here.

In what context would this be used? And to which extent are existing functions not sufficient for doing so?

File: Project.toml [hunk 1]
```diff
@@ -11,6 +11,7 @@ LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 MathOptInterface = "b8f27783-ece8-5eb3-8dc8-9495eed66fee"
 MutableArithmetics = "d8a4904e-b15c-11e9-3269-09a3773c0cb0"
 NaNMath = "77ba4419-2d1f-58cd-9bb1-8ffee604a2e3"
+PrettyTables = "08abe8d2-0d0c-5749-adfa-8a2ac140af0d"
```
Comment (odow) [inline Project.toml]: This also needs an entry in the `[compat]` section.

File: src/JuMP.jl [hunk 2]
```diff
@@ -474,6 +475,38 @@ function solve_time(model::Model)
     return MOI.get(model, MOI.SolveTime())
 end
 
+
+function result!(model::Model; dual_query::Bool = false)
```
Comment (odow) [inline src/JuMP.jl]: What about: `print_solution_summary(model::Model)`?

File: src/JuMP.jl [hunk 3]
```diff
@@ -474,6 +475,38 @@ function solve_time(model::Model)
...
+    data = ["termination status" termination_status(model)]
+
+    if data[2] != MOI.OPTIMIZE_NOT_CALLED
```
Comment (odow) [inline src/JuMP.jl]: Instead of long `if` blocks, prefer
```Julia
if data[2] == MOI.OPTIMIZE_NOT_CALLED
    return pretty_table(data)
end

data = ...
```

**Comment by yashcodes** [comment]
`result!` function basically show the complete result status in a clean tabular form, which also give the option to the user of which data you want to see or not (like the result of dual of the problem)(suggested by [#2191](https://github.com/JuliaOpt/JuMP.jl/issues/2191)).
I agree that the function can be improved, I just wanted reviews on the idea of this kind of function.

**Comment by yashcodes** [comment]
Ok @odow that was a mistake to ping people for reviews, I'll take care of that in future.

File: Project.toml (hunk 1 above)
Comment (yashcodes) [reply Project.toml]: Yes right!!

**Comment by odow** [comment]
So the action item for this is to find what other modeling languages do. e.g., AMPL, GAMS, Excel, Pyomo, CVXPY... They may have good ideas about what information to present.

**Comment by yashcodes** [comment]
The idea behind this PR was simply just to present whatever results a solver can give us by using just one function `JuMP.result(m::Model)`

**Comment by yashcodes** [comment]
I added a few tests, hope it will make clear the aim of the PR.

**Comment by odow** [comment]
Closing because this is not what were are after. We can revisit in a new PR after some discussion in the issue.

**Pull request closed without merging**